seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-jdbc/src/main/java/org/apache/seatunnel/flink/jdbc/source/JdbcSource.java

```diff
@@ -194,10 +195,12 @@ private Map<String, TypeInformation<?>> initTableField(Connection connection, St
             String databaseDialect = connection.getMetaData().getDatabaseProductName();
             PreparedStatement preparedStatement = connection.prepareStatement(selectSql, ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY);
             preparedStatement.setMaxRows(1);
-            ResultSetMetaData rsMeta = preparedStatement.getMetaData();
             try {
+                //support PreparedStatement getMetaData()
+                ResultSetMetaData rsMeta = preparedStatement.getMetaData();
                 return getRowInfo(rsMeta, databaseDialect);
             } catch (SQLException e) {
+                //not support PreparedStatement getMetaData() and use ResultSet getMetaData()
```

Comment:
Not only getMetaData will throw SQLException, maybe other reason will execute this code. If that, the Exception will throw again, please make sure this logic execute in right time.